src/Server.php

```diff
@@ -143,12 +143,18 @@ public function handleRequest(ConnectionInterface $conn, RequestInterface $reque
         $contentLength = 0;
         $stream = new CloseProtectionStream($conn);
         if ($request->hasHeader('Transfer-Encoding')) {
-            $transferEncodingHeader = $request->getHeader('Transfer-Encoding');
-            // 'chunked' must always be the final value of 'Transfer-Encoding' according to: https://tools.ietf.org/html/rfc7230#section-3.3.1
-            if (strtolower(end($transferEncodingHeader)) === 'chunked') {
-                $stream = new ChunkedDecoder($stream);
-                $contentLength = null;
+
+            if (strtolower($request->getHeaderLine('Transfer-Encoding')) !== 'chunked') {
+                $this->emit('error', array(new \InvalidArgumentException('Only chunked-encoding is allowed for Transfer-Encoding')));
+                return $this->writeError($conn, 400);
```

Comment:
Why `400`?

https://tools.ietf.org/html/rfc7230#section-3.3.1 seems to suggest otherwise:

> A server that receives a request message with a transfer coding it
   does not understand SHOULD respond with 501 (Not Implemented).